plugins/in_winevtlog/winevtlog.c

```diff
@@ -612,14 +612,27 @@ struct mk_list *winevtlog_open_all(const char *channels, int read_existing_event
     channel = strtok_s(tmp , ",", &state);
     while (channel) {
         ch = winevtlog_subscribe(channel, read_existing_events, NULL);
-        if (!ch) {
-            flb_free(tmp);
-            winevtlog_close_all(list);
-            return NULL;
+        if (ignore_missing_channels) {
+            if (ch) {
+                mk_list_add(&ch->_head, list);
+            }
```

Comment:
Suppress errors on subscribing non-existent channel is fine. But, we should display warnings when subscribing non-existent channels like as:

```c
else {
  flb_warn("channel %s does not exist", ch);
} 
```

Reply:
Agree on adding that. The log will contain two entries for the same problem though, in the log excerpt posted above, the error can be seen as follows:

> [2022/10/10 19:55:40] [error] [in_winevtlog] cannot subscribe 'BogusChannel' (0)

Reply:
Or, maybe display non-existent logs as debug level at least.

Reply:
This way?
```
else {
  flb_debug("channel %s does not exist", ch);
} 
```